src/jcstress/java/reactor/pool/SimpleDequePoolStressTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2018-Present Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.pool;
+
+import java.time.Duration;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.openjdk.jcstress.annotations.Actor;
+import org.openjdk.jcstress.annotations.Arbiter;
+import org.openjdk.jcstress.annotations.JCStressTest;
+import org.openjdk.jcstress.annotations.Outcome;
+import org.openjdk.jcstress.annotations.State;
+import org.openjdk.jcstress.infra.results.IZI_Result;
+import org.openjdk.jcstress.infra.results.IZL_Result;
+import org.openjdk.jcstress.infra.results.IZ_Result;
+import org.openjdk.jcstress.infra.results.I_Result;
+import org.openjdk.jcstress.infra.results.LI_Result;
+import org.openjdk.jcstress.infra.results.LLI_Result;
+import org.openjdk.jcstress.infra.results.LLLL_Result;
+import org.openjdk.jcstress.infra.results.LLL_Result;
+import org.openjdk.jcstress.infra.results.ZZ_Result;
+
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import static org.openjdk.jcstress.annotations.Expect.*;
+
+public class SimpleDequePoolStressTest {
+
+	@JCStressTest
+	@Outcome(id = {"ACQUIRED 1, 0"}, expect = ACCEPTABLE,  desc = "acquired")
+	@Outcome(id = {"NOT ACQUIRED, 1"}, expect = ACCEPTABLE,  desc = "acquired failed fast during/before destroy")
+	@Outcome(id = {"ACQUIRED 111, 0"}, expect = FORBIDDEN,  desc = "acquired the first resource, before destroy")
+	@Outcome(id = {"ACQUIRED 1111, 0"}, expect = FORBIDDEN,  desc = "acquired the first resource, after destroy")
+	@State
+	public static class MaxPendingAcquireWhileDestroying {
+
+		final AtomicInteger resource = new AtomicInteger(111);
+		final AtomicBoolean firstResourceCreated = new AtomicBoolean();
+
+		final SimpleDequePool<AtomicInteger> pool = PoolBuilder
+				.from(Mono.defer(() -> {
+					if (firstResourceCreated.getAndSet(true)) {
+						return Mono.just(new AtomicInteger(1));
```

Comment:
is it me, or it feels like two tests in one, and that we should rather extract a set of stress tests for `firstResourceCreated`-like scenarios, to not pollute the `MaxPendingAcquireWhileDestroying` ones?

Reply:
I don't think so. The `firstResourceCreated` is there to ensure we produce a (wildly) different `AtomicInteger`, which happens once the first one has been evicted. Having a big difference in the value allows to easily test which one the `acquisition` method saw.

that said I think the test can now be slightly simplified by not requiring the `resource` itself to be a field.

Reply:
Yet it doesn't feel like the creation of resources should be tested in `MaxPendingAcquireWhileDestroying`

Reply:
we don't test it per se, but it is part of the state in which we want our test to occur. reaching `maxPending` is not supposed to be possible if the `acquire` can be served by an available resource. we compete between resource A's eviction and resource B's allocation for the purpose of `acquire`, and test that said `acquire` 1) doesn't fail with a `maxPending` and 2) does indeed trigger an allocation

Reply:
Ok, I think now I understand it better, thanks!